dwave/cloud/client.py

```diff
@@ -311,11 +321,36 @@ def __init__(self, endpoint=None, token=None, solver=None, proxy=None,
         if not endpoint or not token:
             raise ValueError("Endpoint URL and/or token not defined")
 
-        _LOGGER.debug("Creating a client for endpoint: %r", endpoint)
+        _LOGGER.debug(
+            "Creating a client for (endpoint=%r, token=%r, solver=%r, proxy=%r, "
+            "permissive_ssl=%r, request_timeout=%r, polling_timeout=%r, **kwargs=%r)",
+            endpoint, token, solver, proxy, permissive_ssl, request_timeout, polling_timeout, kwargs
+        )
+
+        if solver is None:
+            solver_def = {}
+
+        elif isinstance(solver, collections.Mapping):
+            solver_def = solver
+
+        elif isinstance(solver, six.string_types):
+            # support features dict encoded as JSON in our config INI file
+            # TODO: push this decoding to the config module, once we switch to a
+            #       richer config format (JSON or YAML)
+            try:
+                solver_def = json.loads(solver)
+            except Exception:
+                # unparseable json, assume string name for solver
```

Comment:
Do we want to filter this up to the user, that her json is unparseable, maybe in a log message at least?